tests/checker/telnetserver.py

```diff
@@ -49,7 +49,8 @@ def get_url(self, user=None, password=None):
 
     def setUp (self):
         """Start a new Telnet server in a new thread."""
-        self.port = start_server(self.host, 0)
+        deadline = time.time() + 2 * TIMEOUT
```

Comment:
is that 10 seconds? isn't it a little short for long running tests?

Reply:
Turns out it is!

Reply:
Bumped to 50 seconds.

Reply:
The test should not be "long running": all it does is connect to localhost three times and emit an "exit" command each time. This should take milliseconds!  But that's another issue.

Reply:
Oh, duh!  Linkchecker's telnet:// URL verification logic is
- wait for the string 'login: ' to appear with a 10-second timeout, then send the username
- wait for the string 'Password: ' to appear with a 10-second timeout, then send the password

and the integration test spawns a local telnet server that never sends the login/password prompts, forcing linkchecker to time out.

Reply:
https://github.com/linkchecker/linkchecker/pull/254